File: CS/WindowsApplication1/Helper/MyActionList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.ComponentModel.Design;
using WindowsApplication1.Helper;

namespace WindowsApplication1
{
    public class MyActionList : DesignerActionList
    {
        
   
        public MyUnboundGridHelper Helper
        {
            get
            {
                return Component as MyUnboundGridHelper;
            }
        }

        public MyActionList(IComponent component)
            : base(component)
        {
        }

        public override DesignerActionItemCollection GetSortedActionItems()
        {
            DesignerActionItemCollection items = new DesignerActionItemCollection();
            items.Add(new DesignerActionMethodItem(this,
                                                   "ShowDesigner", "Add unbound data",
                                                   "Data management", "Add unbound data", true));
            return items;
        }


        private void ShowDesigner()
        {
            Helper.ShowDesigner();
        }
    }
}

File: VB/WindowsApplication1/Helper/MyActionList.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.ComponentModel.Design
Imports WindowsApplication1.Helper

Namespace WindowsApplication1
	Public Class MyActionList
		Inherits DesignerActionList


		Public ReadOnly Property Helper() As MyUnboundGridHelper
			Get
				Return TryCast(Component, MyUnboundGridHelper)
			End Get
		End Property

		Public Sub New(ByVal component As IComponent)
			MyBase.New(component)
		End Sub

		Public Overrides Function GetSortedActionItems() As DesignerActionItemCollection
			Dim items As New DesignerActionItemCollection()
			items.Add(New DesignerActionMethodItem(Me, "ShowDesigner", "Add unbound data", "Data management", "Add unbound data", True))
			Return items
		End Function


		Private Sub ShowDesigner()
			Helper.ShowDesigner()
		End Sub
	End Class
End Namespace
